### PR TITLE
chore(cli): move parse repo details to Execute for pipeline init 

### PR DESCRIPTION
**Pull request by efekarakus**
Fixes e2e tests for pipeline

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/pipeline_init_test.go [hunk 1]
```diff
@@ -1085,6 +864,49 @@ func TestInitPipelineOpts_Execute(t *testing.T) {
 			},
 			expectedError: fmt.Errorf("write buildspec to workspace: some error"),
 		},
+		"returns error when repository URL is not from a support git provider": {
```
Comment (huanjani) [inline internal/pkg/cli/pipeline_init_test.go]: nit: 
```suggestion
		"returns error when repository URL is not from a supported git provider": {
```

**Review by iamhopaul123 — APPROVED** [review]

**Review by huanjani — APPROVED** [review]
Thank you for fixing this and cleaning up the unit tests!

File: internal/pkg/cli/pipeline_init_test.go [hunk 2]
```diff
@@ -232,54 +200,7 @@ func TestInitPipelineOpts_Ask(t *testing.T) {
...
-			},
-
-			expectedRepoURL:          codecommitSSHURL,
-			expectedRepoName:         codecommitAnotherRepoName,
-			expectedRepoBranch:       "main",
-			expectedCodeCommitRegion: codecommitRegion,
-			expectedEnvironments:     []string{"test", "prod"},
-			expectedEnvConfigs:       mockEnvConfigs,
-			expectedError:            nil,
+			expectedError: nil,
```
Comment (paragbhingre) [inline internal/pkg/cli/pipeline_init_test.go]: nit: I think `expectedError`is initialized as nil so declaring it here would be redundant.

File: internal/pkg/cli/pipeline_init_test.go [hunk 3]
```diff
@@ -471,12 +286,7 @@ func TestInitPipelineOpts_Ask(t *testing.T) {
 			},
 			mockSessProvider: func(m *mocks.MocksessionProvider) {},
 
-			expectedRepoURL:      githubAnotherURL,
-			expectedGitHubOwner:  githubOwner,
-			expectedRepoName:     githubAnotherRepoName,
-			expectedRepoBranch:   "main",
-			expectedEnvironments: []string{"test", "prod"},
-			expectedError:        nil,
+			expectedError: nil,
```
Comment (paragbhingre) [inline internal/pkg/cli/pipeline_init_test.go]: same as above

File: internal/pkg/cli/pipeline_init_test.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/pipeline_init_test.go]: Done! deleted all `expectedError: nil`

File: internal/pkg/cli/pipeline_init_test.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/pipeline_init_test.go]: Done!

**Pull request merged**